src/VisualStudio/Core/Def/AnalyzerDependency/AnalyzerFileWatcherService.cs

```diff
@@ -6,27 +6,19 @@
 using System.Collections.Generic;
 using System.ComponentModel.Composition;
 using System.IO;
-using System.Linq;
-using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Host.Mef;
-using Microsoft.CodeAnalysis.Shared.Collections;
 using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
-using Microsoft.VisualStudio.LanguageServices.Implementation.TaskList;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
-using Roslyn.Utilities;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation;
 
 [Export(typeof(AnalyzerFileWatcherService))]
-internal sealed class AnalyzerFileWatcherService
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class AnalyzerFileWatcherService(SVsServiceProvider serviceProvider)
 {
-    private static readonly object s_analyzerChangedErrorId = new();
-
-    private readonly VisualStudioWorkspaceImpl _workspace;
-    private readonly HostDiagnosticUpdateSource _updateSource;
```

Comment:
since nothing was listening to the _updateSource anymore, there's no need for the AnalyerFileWatchService to send it any events.  

Note: we may need to do work in LSP pull diags to pull on this type in case it has created an diags.  If that's the case, then we're already in a broken state.  We can see how important that scenario is, and trivially make that work again with LSP pull diags if needed.